frame/evm/src/lib.rs

```diff
@@ -59,6 +63,25 @@ pub trait ConvertAccountId<A> {
 	fn convert_account_id(account_id: &A) -> H160;
 }
 
+/// Truncate the account id, taking the last 160-bit as the Ethereum address.
+pub struct TruncateConvertAccountId;
+
+impl<A: AsRef<[u8]>> ConvertAccountId<A> for TruncateConvertAccountId {
```

Comment:
this might not be particularly secure without a hash first.

Reply:
I changed this to `HashTruncateConvertAccountId` -- it takes a `Hasher`, hash the account id first, and then do the truncate operation.